ImperatorToCK3/Source/Imperator/Countries/CountryFactory.cpp

```diff
@@ -0,0 +1,83 @@
+#include "Log.h"
+#include "ParserHelpers.h"
+#include "CommonRegexes.h"
+#include "CountryFactory.h"
+#include "CountryCurrencies.h"
+#include "CountryName.h"
+
+
+Imperator::Country::Factory::Factory()
+{
+	registerSetter("tag", country->tag);
+	registerKeyword("country_name", [this](std::istream& theStream) {
+		country->name = CountryName(theStream).getName();
+	});
+	registerSetter("flag", country->flag);
+	registerKeyword("country_type", [this](std::istream& theStream) {
+		const auto countryTypeStr = commonItems::getString(theStream);
+		if (countryTypeStr == "rebels")
+			country->countryType = countryTypeEnum::rebels;
+		else if (countryTypeStr == "pirates")
+			country->countryType = countryTypeEnum::pirates;
+		else if (countryTypeStr == "barbarians")
+			country->countryType = countryTypeEnum::barbarians;
+		else if (countryTypeStr == "mercenaries")
+			country->countryType = countryTypeEnum::mercenaries;
+		else if (countryTypeStr == "real")
+			country->countryType = countryTypeEnum::real;
+		else
+		{
+			Log(LogLevel::Error) << "Unrecognized country type: " << countryTypeStr << ", defaulting to real.";
+			country->countryType = countryTypeEnum::real;
+		}
+	});
+	registerKeyword("color", [this](std::istream& theStream) {
+		country->color1 = commonItems::Color::Factory().getColor(theStream);
+	});
+	registerKeyword("color2", [this](std::istream& theStream) {
+		country->color2 = commonItems::Color::Factory().getColor(theStream);
+	});
+	registerKeyword("color3", [this](std::istream& theStream) {
+		country->color3 = commonItems::Color::Factory().getColor(theStream);
+	});
+	registerKeyword("currency_data", [this](std::istream& theStream) {
+		const CountryCurrencies currenciesFromBloc(theStream);
+		country->currencies.manpower = currenciesFromBloc.getManpower();
+		country->currencies.gold = currenciesFromBloc.getGold();
+		country->currencies.stability = currenciesFromBloc.getStability();
+		country->currencies.tyranny = currenciesFromBloc.getTyranny();
+		country->currencies.war_exhaustion = currenciesFromBloc.getWarExhaustion();
+		country->currencies.aggressive_expansion = currenciesFromBloc.getAggressiveExpansion();
+		country->currencies.political_influence = currenciesFromBloc.getPoliticalInfluence();
+		country->currencies.military_experience = currenciesFromBloc.getMilitaryExperience();
+	});
+	registerKeyword("capital", [this](std::istream& theStream) {
+		auto capitalLongLong = commonItems::getULlong(theStream);
```

Comment:
Just 'capital' is fine

Reply:
Renamed to `capitalProvID` for less confusion with the member.